tests/foreman/longrun/test_capsule_loadbalancer.py

```diff
@@ -16,74 +16,216 @@
+        source=get_data_file('haproxy.cfg'), destination='/etc/haproxy/haproxy.cfg'
+    )
+    haproxy.execute(
+        f'sed -i -e s/CAPSULE_1/{setup_capsules["capsule_1"].hostname}/g '
+        f' --e s/CAPSULE_2/{setup_capsules["capsule_2"].hostname}/g '
+        f' /etc/haproxy/haproxy.cfg'
+    )
+    haproxy.execute('systemctl restart haproxy.service')
+    haproxy.execute('mkdir /var/lib/haproxy/dev')
+    haproxy.session.sftp_write(
+        source=get_data_file('99-haproxy.conf'), destination='/etc/rsyslog.d/99-haproxy.conf'
+    )
+    haproxy.execute('setenforce Permissive')
+    result = haproxy.execute('systemctl restart haproxy.service rsyslog.service')
+    assert result.status == 0
+
+    return {'haproxy': haproxy}
+
+
+@pytest.fixture(scope='module')
+def loadbalancer_setup(
+    module_org, content_for_client, setup_capsules, module_target_sat, setup_haproxy
+):
+    """Just to keep the tests arguments clean. Yes it's redundant"""
+    return {
+        'module_org': module_org,
+        'content_for_client': content_for_client,
+        'setup_capsules': setup_capsules,
+        'module_target_sat': module_target_sat,
+        'setup_haproxy': setup_haproxy,
+    }
+
+
+@pytest.mark.dependency()
 @pytest.mark.tier1
-def test_manually_register_client_using_ak():
+def test_loadbalancer_register_client_using_ak_to_capsule(loadbalancer_setup, rhel7_contenthost):
     """Register the client using ak to the capsule
 
     :id: 7318c380-e149-11ea-9b17-4ceb42ab8dbc
 
-    :Steps: run `subscription-manager register --org=Your_Organization \
+    :Steps:
+        1. run `subscription-manager register --org=Your_Organization \
             --activationkey=Your_Activation_Key \
             --serverurl=https://loadbalancer.example.com:8443/rhsm \
             --baseurl=https://loadbalancer.example.com/pulp/repos`
 
     :expectedresults: The client should be registered to one of the capsules
 
+    :CaseLevel: Integration
     """
-    pass
-
-
-@pytest.mark.stubbed
+    url = f'https://{loadbalancer_setup["setup_haproxy"]["haproxy"].hostname}'
+    server_url = f'{url}:8443/rhsm'
+    base_url = f'{url}/pulp/content/'
+    result = rhel7_contenthost.download_install_rpm(
+        repo_url=f'{url}/pub', package_name='katello-ca-consumer-latest.noarch'
+    )
+    assert result.status == 0
+    rhel7_contenthost.register_contenthost(
+        org=loadbalancer_setup['module_org'].label,
+        activation_key=loadbalancer_setup['content_for_client']['client_ak'].name,
+        serverurl=server_url,
+        baseurl=base_url,
+    )
+    result = rhel7_contenthost.execute('rpm -qa |grep katello-ca-consumer')
+    assert (
+        loadbalancer_setup['setup_capsules']['capsule_1'].hostname
+        or loadbalancer_setup['setup_capsules']['capsule_2'].hostname in result.stdout
+    )
+
+
+@pytest.mark.dependency(depends=['test_loadbalancer_register_client_using_ak_to_capsule'])
 @pytest.mark.tier1
 def test_list_repolist():
```

Comment:
I wonder this test is not marked as stubbed :)

Mark it please !

Reply:
I was testing the `depend` in pytest :)